### PR TITLE
[function](string_function) add new string function 'not_null_or_empty'

### PR DESCRIPTION
**Pull request by zy-kkk**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [x] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [x] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by BiteTheDDDDt — CHANGES_REQUESTED** [review]
Need add test

File: be/src/vec/functions/function_string.h [hunk 1]
```diff
@@ -420,6 +420,43 @@ class FunctionNullOrEmpty : public IFunction {
...
+    bool use_default_implementation_for_constants() const override { return true; }
+
+    Status execute_impl(FunctionContext* context, Block& block, const ColumnNumbers& arguments,
+                        size_t result, size_t input_rows_count) override {
+        auto res_map = ColumnUInt8::create(input_rows_count, 0);
+
+        auto column = block.get_by_position(arguments[0]).column;
+        if (auto* nullable = check_and_get_column<const ColumnNullable>(*column)) {
+            column = nullable->get_nested_column_ptr();
+            res_map->get_data();
```
Comment (BiteTheDDDDt) [inline be/src/vec/functions/function_string.h:444]: We better check nullmap when input column is nullable.

Comment (BiteTheDDDDt) [inline be/src/vec/functions/function_string.h:444]: line 444 seems unused

File: docs/zh-CN/docs/sql-manual/sql-functions/string-functions/not_null_or_empty.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+`BOOLEAN NOT_NULL_OR_EMPTY (VARCHAR str)`
+
+如果字符串为空字符串或者NULL，返回false。否则，返回true。
+
+### example
+
+```
+MySQL [(none)]> select not_null_or_empty(null);
++---------------------+
+| null_or_empty(NULL) |
```
Comment (BiteTheDDDDt) [inline docs/zh-CN/docs/sql-manual/sql-functions/string-functions/not_null_or_empty.md:40]:
```suggestion
| not_null_or_empty(NULL) |
```

Comment (zy-kkk) [reply docs/zh-CN/docs/sql-manual/sql-functions/string-functions/not_null_or_empty.md]: done

**Comment by zy-kkk** [comment]
> Need add test

done

File: be/src/vec/functions/function_string.h (hunk 1 above)
Comment (BiteTheDDDDt) [reply be/src/vec/functions/function_string.h]: I mean, we can't guarantee that when the xth element of a ColumnNullable is null, it must have a length of 0.

**Review by BiteTheDDDDt — APPROVED** [review]
LGTM

**Pull request merged**